src/contexts/addresses.tsx

```diff
@@ -456,20 +459,18 @@ export const AddressesContextProvider: FC<{ overrideContextValue?: PartialDeep<A
     return () => {
       clearInterval(interval)
     }
+  }, [
+    addressesState,
+    currentNetwork,
+    fetchAndStoreAddressesData,
+    addressesOfCurrentNetwork,
+    addressesWithPendingSentTxs
+  ])
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [checkForNewPending])
-
-  useEffect(() => {
-    const interval = setInterval(async () => {
-      await fetchPendingTxs(addressesOfCurrentNetwork)
-      setCheckForNewPending({})
-      // MinimumWaitTime + HalfAvgBlockTime
-    }, 1000 + (1000 * 60) / 2)
-    return () => {
-      clearInterval(interval)
-    }
-  }, [addressesOfCurrentNetwork, fetchPendingTxs])
```

Comment:
Since we are not checking for new confirmed transactions unless the user hits the "Refresh" button, I don't see a reason to do that with unconfirmed txs. I have updated the refresh function to also check for unconfirmed.